src/main/java/utilities/OS.java

```diff
@@ -1,21 +1,38 @@
 package utilities;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 public enum OS {
     WINDOWS,
     LINUX,
+    MAC_OS_X,
     UNKNOWN;
 
+    private static Logger logger = LogManager.getLogger(OS.class);
+
+    /**
+     * Detects the OS of system in which this method is called.
+     * @return The detected OS. If it's unable to detect, it will return UNKNOWN.
+     */
     public static OS getOs() {
-        String os = System.getProperty("os.name").toUpperCase();
-        System.out.println(os.contains("LINUX"));
-        if(os.contains("LINUX")) {
-            return LINUX;
+        String osString = System.getProperty("os.name")
+                .toUpperCase();
+        logger.debug("OS string: " + osString);
+        OS detectedOS;
+        if(osString.contains("LINUX")) {
+            detectedOS = LINUX;
+        }
+        else if(osString.contains("WINDOWS")) {
+            detectedOS = WINDOWS;
         }
-        else if(os.contains("WINDOWS")) {
-            return WINDOWS;
+        else if(osString.contains("MAC OS X")) {
```

Comment:
Curious, will this work on all mac versions ? Online, i see, they check presence only for "MAC" in it

Reply:
I tested it on the GitHubAction runner, it works. We can fix it later if poses an issue to certain Mac versions.